src/Operation/GroupBy.php

```diff
@@ -73,7 +73,7 @@ static function (array $collect, $value, $key) use ($callback): array {
 
                 /** @var Closure(Iterator<TKey, T>): Generator<int, list<T>> $pipe */
                 $pipe = Pipe::of()(
-                    FoldLeft::of()($reducerFactory($callable))([]),
+                    Reduce::of()($reducerFactory($callable))([]),
```

Comment:
you mention in the commit message that this is more performant. Can you clarify why that is?

Also, maybe we could have this refactor in a dedicated PR?

Reply:
If we could draw a tree of dependency for each Operation (_actually it could be a very nice idea_), we can see that

`FoldLeft`: [ [Reduction, Prepend], Last ]
`Reduce`: [Reduction, Last]

The `Reduce` operation does not need the `Prepend` operation and then, it's faster.

Reply:
nice explanation, thanks. Indeed would be cool to have some sort of dependency tree.

I've used this before to enforce dependencies, but we can also use it to visualise them: https://github.com/qossmic/deptrac

Reply:
I created the discussion to not forget this in the void! #149